tests/types/test-time64.py

```diff
@@ -488,3 +488,78 @@ def test_cast_string_to_time64(ttype):
     assert_equals(DT, dt.Frame([d(2001, 7, 11, 12, 5, 59, 999000),
                                 d(2002, 10, 4, 23, 0, 1),
                                 None, None]))
+
+
+
+
+#-------------------------------------------------------------------------------
+# Type casts from `time64` type
+#-------------------------------------------------------------------------------
+
+def test_cast_time64_to_void():
+    DT = dt.Frame([d(2111, 11, 11, 11, 11, 11, 111111)])
+    DT[0] = dt.Type.void
+    assert_equals(DT, dt.Frame([None]))
+
+
+def test_cast_time64_to_int64():
+    DT = dt.Frame([d(1970, 1, 1, 0, 0, 0),
+                   d(1970, 1, 1, 0, 0, 1),
+                   d(1970, 1, 2, 0, 0, 0, 1),
+                   d(1969, 12, 31, 23, 59, 59, 999999),
+                   None])
+    DT[0] = dt.Type.int64
+    assert_equals(DT, dt.Frame([0,
+                                1000000000,
+                                86400000001000,
+                                -1000,
+                                None]))
+
+
+def test_cast_time64_to_double():
+    DT = dt.Frame([d(2013, 10, 5, 11, 23, 45, 157839)])
+    DT[0] = float
+    assert_equals(DT, dt.Frame([1380972225157839000.0]))
+
+
+def test_cast_time64_to_date32():
+    from datetime import date
+    DT = dt.Frame([d(2091, 11, 28, 15, 51, 27, 310000),
+                   d(1970, 1, 3, 21, 5, 2, 475000),
+                   d(1969, 12, 31, 23, 59, 59, 999999),
+                   d(1969, 12, 31, 0, 0, 0),
+                   d(1962, 4, 12, 3, 52, 27, 458000),
+                   None])
+    assert DT.type == dt.Type.time64
+    DT[0] = dt.Type.date32
+    assert_equals(DT, dt.Frame([date(2091, 11, 28),
+                                date(1970, 1, 3),
+                                date(1969, 12, 31),
+                                date(1969, 12, 31),
+                                date(1962, 4, 12),
+                                None]))
+
+
+@pytest.mark.parametrize('st', [dt.str32, dt.str64])
+def test_cast_time64_to_string(st):
+    DT = dt.Frame([d(2091, 11, 28, 15, 51, 27, 310000),
+                   d(1970, 1, 3, 21, 5, 2, 475000),
+                   d(1969, 12, 31, 23, 59, 59, 999999),
+                   d(1969, 12, 31, 0, 0, 0),
+                   d(1962, 4, 12, 3, 52, 27, 458000),
+                   None])
+    DT[0] = st
+    assert_equals(DT, dt.Frame(["2091-11-28T15:51:27.31",
+                                "1970-01-03T21:05:02.475",
+                                "1969-12-31T23:59:59.999999",
+                                "1969-12-31T00:00:00",
+                                "1962-04-12T03:52:27.458",
+                                None], stype=st))
```

Comment:
What is the reason we're using "T" as the date/time separator and not just a space?

Reply:
It's in the ISO-8601 standard: https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations

Reply:
Yes, but the space seems to be more readable and is allowed through the RFC 3339:

```
Separating date and time parts with other characters such as space is not allowed in ISO 8601, but allowed in its profile RFC 3339.[34]

...

RFC 3339". 2002-07-01. "5.6. NOTE: ISO 8601 defines date and time separated by "T". Applications using this syntax may choose, for the sake of readability, to specify a full-date and full-time separated by (say) a space character."
```

Reply:
ISO = International Standardization Organization
RFC = Request For Comments
So, I take ISO to be more authoritative than RFC. While I agree that space separator is easier on the eyes, the older ISO standard did allow both a "T" and a space here, but in the end, they changed to forbid the space separator. This means, they must have had a good reason for such a change.

Reply:
Frankly I can't find that reason, but what I found is this quote from  ISO-8601:

```
NOTE By mutual agreement of the partners in information interchange, the character [T] may be omitted in applications where there is no risk of confusing a date and time of day representation with others defined in this International Standard.
```

So it seems that they wanted to prevent confusion between reading in date and time. But this quote is more about fully eliminating the `T`, if `T` is replaced by a space I don't see a source of confusion. May be there is a reason, just can't get it. Having `T` instead of a space looks kind of messy imho.

Reply:
I don't know, the EDTF standard also says T: https://www.loc.gov/standards/datetime/

Reply:
Even god-forsaken `ГОСТ Р 7.0.64-2018` specifies using T: https://docs.cntd.ru/document/1200159341
What other standards are out there?

Reply:
Not sure :) Do we also require the `T` separator when parsing strings into time? It seems that pandas, for example, allows both `T` and ` ` separators.

Reply:
when parsing we allow both space and T